### PR TITLE
blog: Devtools deep dive

### PR DESCRIPTION
**Pull request by eerii**
This is a very first draft of the blog post about the devtools support. I tried not making it too long, but I can add more content if it is needed. I'm still a bit hesitant on some sections, specially the first paragraph and the console one, as well as the summary, but I will update them if I can think of something better. I also need to add the images and links.

@delan please make any changes you consider necessary and give any feedback on what needs to be different. Thanks!

### CODE REVIEW

**Comment by delan** [comment]
Hi, is this ready for review? No rush, just wondering since it’s still marked as draft :)

**Comment by eerii** [comment]
Hi, sorry about the delay, this week was a bit chaotic. I wanted to re-read it once more and add images before marking it for review. I still need to add the PR links, is there any automated way to do that? And maybe the images should be a tad larger, what do your think?

**Comment by delan** [comment]
No worries, take all the time you need! I partially automate those links by typing the number, exactly like you did, then [using find-and-replace](https://github.com/servo/servo.org?tab=readme-ov-file#how-to-linkify-github-handles-and-pull-requests-when-finishing-a-post) right before publishing.

I think given your original images, making them a bit bigger on the page would be nice. How legible we can make them depends on their width though, so for mobile that doesn’t really help:

image

I wonder if we can recompose the figures to make them narrower? The side-by-side figure could be top-and-bottom or overlapping (🗗), and we could resize the windows to reduce the empty space make the panels stack vertically. Some examples using devtools in Firefox:

image

image

**Comment by delan** [comment]
Much better! I’ve done some minor copyediting in 73da0a0a5180d91e016c7a3b571543e9d5a06a08. I also fixed the figures so they shrink properly on viewports narrower than 33em in f176dc4f07eb4bfb0350e67f0b62ed1a2f22de28, by pulling in the latest per-post styles. I think we needed the `._figr { max-width: 100%; }` part for the `._figr > a > img { max-width: 100%; }` to work.

I’ll do a second pass shortly :)

File: _posts/2024-09-01-devtools.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+---
+
+<figure class="_figr"><a href="{{ '/img/blog/devtools-inspector.png' | url }}"><img src="{{ '/img/blog/devtools-inspector.png' | url }}"
+    alt="On the left, it shows the DOM inspector with the tree view, CSS list and computed properties views. On the right is servoshell with servo.org opened."></a>
+<figcaption>The HTML and CSS inspector is able to display the DOM elements and their attributes and CSS properties.</figcaption></figure>
+
+<span class=_floatmin></span>Servo has been working on improving our [Firefox devtools](https://firefox-source-docs.mozilla.org/devtools-user) support as part of the [Outreachy](https://www.outreachy.org) internship program since June, and we're thrilled to share significant progress.
+
+Devtools are a set of **web developer tools inside Firefox** that allows you to examine, edit, and debug HTML, CSS, and JavaScript.
+Servo leverages existing work there and uses the same open protocol to inspect its own websites.
```
Comment (delan) [inline _posts/2024-09-01-devtools.md:17]: this reads a little odd because devtools like this exist in all of the major browsers. i wonder if we can reword it to move the Firefox mention to the second sentence, so we go from general (devtools are browser web developer tools) to specific (we use the Firefox devtools)?

Comment (delan) [reply _posts/2024-09-01-devtools.md:17]: it may also be worth adding something about how Firefox lets you point their devtools to another Firefox, like on a mobile device, which is why they have a protocol we can use too.

**Review by delan — COMMENTED** [review]
I’ve made some bigger changes in 95df3dfa8f3ca839b297de8f7df834ae9dc4400b and 1e7ab380ff222c6e497a86a7154d72d3c2eb1086, reordering the end of the post and giving the post a (hopefully) more engaging title and summary.

One last suggestion, but aside from that it looks good:

File: _posts/2024-09-01-devtools.md (hunk 1 above)
Comment (eerii) [reply _posts/2024-09-01-devtools.md]: Thank you for all of the changes, it reads great now. How about something like this for this paragraph?

Comment (delan) [reply _posts/2024-09-01-devtools.md]: looks good!

**Review by delan — APPROVED** [review]

File: _posts/2024-09-01-devtools.md (hunk 1 above)
Comment (delan) [reply _posts/2024-09-01-devtools.md]: looks good, thanks!

**Comment by delan** [comment]
Thanks for all your hard work! We’ll publish this on Monday :)

**Comment by eerii** [comment]
Thank you so much for all of your help with this! c:
